### ISSUE TITLE
`Error rendering buttons of the cell toolbar` when deleting cells in a notebook

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#14380]
## Description

Using `4.0.0b1` on Binder can lead to the following errors being logged in the dev tools console:

```
Error rendering buttons of the cell toolbar:  TypeError: t.layout is null
```

## Reproduce

- Create a notebook
- Create cells
- Select cells
- Delete them with the cut toolbar item

https://user-images.githubusercontent.com/591645/232113598-020ef944-07e6-4090-ae72-a46e0cf40f2a.mp4

## Expected behavior

No error in the dev tools console.

## Context

- JupyterLab `4.0.0b1` on Binder: https://gist.github.com/jtpio/4e9a3d435d7c3950dc4150c10abbb1d7

**Comment by krassowski** [comment]
Switching cell type produces a similar error:

```
Error rendering buttons of the cell toolbar:  TypeError: Cannot read properties of null (reading 'insertWidget')
```

or in Firefox:

```
Error rendering buttons of the cell toolbar:  TypeError: can't access property "insertWidget", t.layout is null
```

![switching cell type](https://user-images.githubusercontent.com/5832902/232255008-772e3ebc-26d5-4ce2-ac96-325e0bf1be26.gif)

**Comment by JasonWeill** [comment]
May be fixed if #14139 is done

**Comment by JasonWeill** [comment]
As discussed at the JupyterLab weekly meeting, this is a bug we should fix in 4.0.x, but is not a release blocker for 4.0.0.

**Comment by krassowski** [comment]
I encountered something similar while porting the vim extension; it appears that when cell type is being switch the `activeCellChanged` signal emits with the old cell which is already disposed. It should not do that IMO.
